specifications/android.md

```diff
@@ -82,32 +87,38 @@ There is a third action the connector SHOULD handle:
 ### org.unifiedpush.android.connector.NEW_ENDPOINT
 
 The distributor MUST send this action to the registered application to confirm the registration of an end user application, when a registered application send again an action with the intent org.unifiedpush.android.distributor.REGISTER and a valid token, or when the endpoint change with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* endpoint: the endpoint
+* token (String): the token supplied by the end user application during registration
+* endpoint (String): the endpoint
 
 ### org.unifiedpush.android.connector.REGISTRATION_FAILED
 
 The distributor MUST send this action to the registered application if the token is already registered for another application or if the registration can not be processed (for instance when the distributor is not connected to its provider server) with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* message: this extra MAY be sent to gives an error message
+* token (String): the token supplied by the end user application during registration
+* message (String): this extra MAY be sent to gives an error message
 
 The connector MUST change the connection token received with this action for the next registration.
 
 ### org.unifiedpush.android.connector.MESSAGE
 
 The distributor MUST send this action to the registered application to forward a push message received from the provider to the end user application.
 
-It MUST be send with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* message: the push message sent by the application server. It MUST be the raw POST data received by the rewrite proxy.
+It MUST be send with the following extra:
+* token (String): the token supplied by the end user application during registration
+
+It MUST be send with at least one of the 2 following extras:
+* message (String): the push message sent by the application server, as a string. It MUST be setted if the connector has been registered without support for bytesMessage feature. If setted, it MUST be the raw POST data received by the rewrite proxy.
```

Comment:
It is ambiguous if the distributor has to set this if connected with the feature.

I would say: It MUST be set if the bytesMessage feature was not present, and it MAY be present if it was present during registration.

Grammar: "If set, ..." Not "If setted"